core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -95,12 +95,13 @@ public CompletableFuture<Map<TopicIdPartition, ShareFetchResponseData.PartitionD
         String groupId,
         String memberId,
         FetchParams fetchParams,
-        List<TopicIdPartition> topicIdPartitions) {
+        List<TopicIdPartition> topicIdPartitions,
+        Map<TopicIdPartition, Integer> partitionMaxBytes) {
```

Comment:
nit: the name seems better earlier `topicIdPartitions` because it's not just partitionMaxBytes we needed but the partitions information for which the data is to be fetched. Though it's fine to keep the map for now but if you do see some other partition level data that should flow from KafkaApis to SharePartitionManager then better would be to keep a struct, if not then let's keep the map itself.

Having said that, there was a requirement where we did say to shuffle the partitions in the next fetch request. So ids the map implementation in KafkaApis ordered?

Reply:
Regarding first point, I had checked. Its only PartitionMaxBytes that we need. We do not need anything else from the topic partition perspective.
Regarding second point, I think you understood it [here](https://github.com/confluentinc/kafka/pull/1092#discussion_r1524663541)